parsec/core/gui/custom_dialogs.py

```diff
@@ -53,9 +66,15 @@
     QEventLoop,
 )
 
+P = ParamSpec("P")
+R = TypeVar("R")
+T_CLASS = TypeVar("T_CLASS")
```

Comment:
`typing.Type` is most likely enough here.

Reply:
Mypy complains about missing type parameters for generic type `Type`